### DISCUSSION TITLE
Help setting up HLS with GHCup-provided GHC?

### DISCUSSION THREAD

**Discussion by gwerbin** [haskell/haskell-language-server#3884] [Q&A] [answered]
I installed GHC, Stack, Cabal, and HLS using GHCUp, but HLS cannot find the GHC version that was installed by GHCUp. Apparently, GHC is not available as `ghc`, but only as `ghc-9.4.7` (in `~/.ghcup/bin`), and HLS doesn't know to look for that particular binary. There is a "plain" `ghc` binary in `~/.ghcup/ghc/9.4.7/bin/`, but it doesn't seem like I'm expected to add that directory to `$PATH`.

Is this the expected behavior, or did I miss some configuration step? Is there some kind of project config file I can use to specify the GHC compiler path that HLS should use?

**Comment by fendor** [comment, marked as answer]
Hi!

Make sure you set the ghc version you want to use, either via `ghcup set ghc 9.4.7` or in `ghcup tui`, make sure you have two ticks:

![image](https://github.com/haskell/haskell-language-server/assets/8463814/e5caddbf-091b-457a-9ef0-97806189f93d)


Ignore the prerelease stuff, but you can see that `9.6.3` has two ticks while other ghc versions have only one. The GHC version that is currently set is the one `ghc` will point to. So in my case, 
```
> ghc --version
The Glorious Glasgow Haskell Compilation System, version 9.6.3
```

**Reply by gwerbin** [reply]
Thanks! Was that in the beginner setup instructions? I must have missed it.

**Reply by fendor** [reply]
Hm, I thought the GHCup installer should be doing that automatically... Maybe you aborted too early or something? I am not sure, I thought it would be part of the installation instructions...
